main/lsp/LSPFileUpdates.h

```diff
@@ -66,18 +66,20 @@ class LSPFileUpdates final {
     // As written, it's possible to mutate the files, but that's not intentional.
     static FastPathFilesToTypecheckResult
     fastPathFilesToTypecheck(const core::GlobalState &gs, const LSPConfiguration &config,
-                             const std::vector<std::shared_ptr<core::File>> &updatedFiles);
+                             const std::vector<std::shared_ptr<core::File>> &updatedFiles,
+                             bool isNoopUpdateForRetypecheck);
 
     // Overload because sometimes we have to look up the old file's hash in GlobalState (when
     // running on the typechecker thread), and sometimes we have to look it up in evictedFiles (when
     // running on the indexer thread).
     static FastPathFilesToTypecheckResult
     fastPathFilesToTypecheck(const core::GlobalState &gs, const LSPConfiguration &config,
                              const std::vector<std::shared_ptr<core::File>> &updatedFiles,
-                             const UnorderedMap<core::FileRef, std::shared_ptr<core::File>> &evictedFiles);
+                             const UnorderedMap<core::FileRef, std::shared_ptr<core::File>> &evictedFiles,
+                             bool isNoopUpdateForRetypecheck);
 
-    FastPathFilesToTypecheckResult fastPathFilesToTypecheck(const core::GlobalState &gs,
-                                                            const LSPConfiguration &config) const;
+    FastPathFilesToTypecheckResult fastPathFilesToTypecheck(const core::GlobalState &gs, const LSPConfiguration &config,
+                                                            bool isNoopUpdateForRetypecheck) const;
```

Comment:
Can you remind me whether we have a path towards not needing to use this boolean, or whether it's a parameter we expect to be here forever?

I had tried pretty hard to not need to know whether the edit was a retypecheck edit or whether it was a normal edit throughout most of the LSP code, so I would like to understand the context in case we ever do something in the future which would let us remove this boolean.